src/Integrations/Integrations/Laravel/LaravelIntegration.php

```diff
@@ -288,6 +311,125 @@ function ($exceptionHandler, $scope, $args) use ($rootSpan, $integration) {
+                    return;
+                }
+
+                $metadata = [];
+                if (isset($user['name'])) {
+                    $metadata['name'] = $user['name'];
+                }
+                if (isset($user['email'])) {
+                    $metadata['email'] = $user['email'];
+                }
+
+                \datadog\appsec\track_user_login_success_event($user->getAuthIdentifier(), $metadata, true);
+            }
+        );
+
+        // Used by Laravel < 5.0
+        \DDTrace\hook_method(
+            'Illuminate\Auth\Guard',
+            'setUser',
+            function ($This, $scope, $args) use ($rootSpan, $integration) {
+                if (!function_exists('\datadog\appsec\track_user_login_success_event'))
+                {
+                    return;
+                }
+                if (!isset($args[0])) {
+                    return;
+                }
+                $user = $args[0];
+                $authClass = 'Illuminate\Auth\UserInterface';
+                if (!$user || !($user instanceof $authClass)) {
+                    return;
+                }
+
+                $metadata = [];
+                if (isset($user['name'])) {
+                    $metadata['name'] = $user['name'];
+                }
+                if (isset($user['email'])) {
+                    $metadata['email'] = $user['email'];
+                }
+
+                \datadog\appsec\track_user_login_success_event($user->getAuthIdentifier(), $metadata, true);
+            }
+        );
+
+        // Used by Laravel < 5.0
+        \DDTrace\hook_method(
+            'Illuminate\Auth\Guard',
+            'attempt',
+            null,
+            function ($This, $scope, $args, $loginSuccess) use ($rootSpan, $integration) {
+                if ($loginSuccess) {
+                    return;
+                }
+
+                if (!function_exists('\datadog\appsec\track_user_login_failure_event'))
+                {
+                    return;
+                }
+                \datadog\appsec\track_user_login_failure_event(null, false, [], true);
+            }
+        );
+
+        \DDTrace\hook_method(
+            'Illuminate\Auth\Events\Registered',
+            '__construct',
+            null,
+            function ($This, $scope, $args) use ($rootSpan, $integration) {
+                if (!function_exists('\datadog\appsec\track_user_signup_event'))
+                {
+                    return;
+                }
+                if (!isset($args[0])) {
+                    return;
+                }
+                $user = $args[0];
+                $authClass = '\Illuminate\Contracts\Auth\Authenticatable';
+                if (!$user || !($user instanceof $authClass)) {
+                    return;
+                }
+
+                $metadata = [];
+                \datadog\appsec\track_user_signup_event($user->getAuthIdentifier(), $metadata, true);
```

Comment:
**nitpick (non-blocking):** this is just a coding style thing
```suggestion
                if (!function_exists('\datadog\appsec\track_user_signup_event') ||
                    !isset($args[0]) ||
                    !$args[0] ||
                    !($args[0] instanceof \Illuminate\Contracts\Auth\Authenticatable))
                {
                    return;
                }
                \datadog\appsec\track_user_signup_event($args[0]->getAuthIdentifier(), [], true);
```
I do not have any hard feelings about this and I do not block the PR because of this, If you like to change this, you could also do in a separate PR later.

Reply:
Since I was doing the other changes on this file I have changed this and all other functions I added :)